packages/datadog-plugin-paperplane/src/index.js

```diff
@@ -53,24 +53,53 @@ const wrapRoutes = tracer => routes => handlers => {
   return routes(traced)
 }
 
-function patch (paperplane, tracer, config) {
-  config = web.normalizeConfig(config)
-
-  if (tracer._logInjection) {
-    this.wrap(paperplane, 'logger', wrapLogger(tracer))
+module.exports = [
+  {
+    name: 'paperplane',
+    versions: ['>=2.3.2'],
+    file: 'lib/logger.js',
+    patch (exports, tracer) {
+      if (tracer._logInjection) {
+        this.wrap(exports, 'logger', wrapLogger(tracer))
+      }
+    },
+    unpatch (exports) {
+      this.unwrap(exports, 'logger')
+    }
+  },
+  {
+    name: 'paperplane',
+    versions: ['>=2.3.2'],
+    file: 'lib/mount.js',
+    patch (exports, tracer, config) {
+      config = web.normalizeConfig(config)
+      this.wrap(exports, 'mount', wrapMount(tracer, config))
+    },
+    unpatch (exports) {
+      this.unwrap(exports, 'mount')
+    }
+  },
+  {
+    name: 'paperplane',
+    versions: ['>=2.3.2'],
+    file: 'lib/routes.js',
+    patch (exports, tracer) {
+      this.wrap(exports, 'routes', wrapRoutes(tracer))
+    },
+    unpatch (exports) {
+      this.unwrap(exports, 'routes')
+    }
+  },
+  {
+    name: 'paperplane',
+    versions: ['2.3.0 - 2.3.1'],
```

Comment:
It looks like both `lib/routes.js` and `lib/mount.js` were available in these versions as well. Could this block simply be removed and the requirements above reduced?

Reply:
I originally thought I'd do that, but it won't work.  The files themselves were available then, but internally they exported as the default export, rather than a named export.  For example:

**`lib/routes.js` in 2.3.1:**
```js
module.exports = curry(routes)
```

**`lib/routes.js` in 2.3.2:**
```js
exports.routes = curry(routes)
```

The named exports in 2.3.2 was done to support internal patching, and it's not supported prior to that.  So I had to patch them differently for the earlier versions.

Reply:
We went ahead and did named exports for all internal functions, rather than just `logger`, to avoid any more future bumps to support patching in Datadog or any other third-party library.

Reply:
But then doesn't the opposite apply? (using the named export from the library itself instead of the internal files).

It's also possible to overwrite the default export using the instrumenter, but it comes with a few limitations so the additional block is probably better to avoid that.

I'm just trying to reduce the number of patches if possible 😄

Reply:
If we patch it internally, then the externally exported reference is also patched.  `shimmer` and `require-in-the-middle` do a good job of allowing that.

And just to be sure:  the patch is done internally for `v2.3.2`, but in the tests [I'm still testing the externally exported functions](https://github.com/DataDog/dd-trace-js/pull/633/files#diff-b5639a388194ae9b8716861596960d06R532), and it works just fine.

Reply:
What I meant was, why not just always patch externally as it was before? In general that is the preferred method so that the instrumentation continues to work even if the internal structure changes. Instrumenting internal modules is a last resort for when what we need to patch is not exported.

Reply:
I totally understand.  But these are first-class functions we are patching/replacing, not prototype methods.  So if I patched just the external export, it would just be patching `require('paperplane').logger`.  Internally, when `lib/mount.js` uses the logger, it would not be patched, because it uses `const { logger: defaultLogger } = require('./logger')`.

I get it.  I would rather patch just the interface.  But in this case, if I just patch externally, and I don't supply a custom logger, it would use the default, and no logs would be injected.

Reply:
Fair enough.